test/lit/passes/heap2local.wast

```diff
@@ -0,0 +1,1527 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py and should not be edited.
+;; (remove-unused-names allows the pass to see that blocks flow values)
+;; RUN: wasm-opt %s -all --remove-unused-names --heap2local -S -o - | filecheck %s
+
+(module
+  (type $struct.A (struct (field (mut i32)) (field (mut f64))))
+
+  (type $struct.packed (struct (field (mut i8))))
+
+  (type $struct.nondefaultable (struct (field (rtt $struct.A))))
+
+  (type $struct.recursive (struct (field (mut (ref null $struct.recursive)))))
+
+  (type $struct.nonnullable (struct (field (ref $struct.A))))
+
+  ;; CHECK:      (func $simple
+  ;; CHECK-NEXT:  (local $0 i32)
+  ;; CHECK-NEXT:  (local $1 f64)
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (block (result (ref $struct.A))
+  ;; CHECK-NEXT:    (local.set $0
+  ;; CHECK-NEXT:     (i32.const 0)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:    (local.set $1
+  ;; CHECK-NEXT:     (f64.const 0)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:    (struct.new_default_with_rtt $struct.A
+  ;; CHECK-NEXT:     (rtt.canon $struct.A)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $simple
+    ;; Other passes can remove such a simple case of an unneeded allocation, and
+    ;; we do not bother with it, as there are struct get/set operations that we
```

Comment:
Oh, good point, heh, not sure how I missed that on the very first testcase...

The logic to avoid trivial things was wrong. Reading it, it's not easy to get right. I removed it so we do handle it here. There's no harm really.